conf/cache.c

```diff
@@ -102,6 +103,51 @@ M0_INTERNAL bool m0_conf_cache_contains(struct m0_conf_cache *cache,
 	return ret;
 }
 
+M0_INTERNAL struct m0_conf_obj *
+m0_conf_cache_lookup_dynamic_fid(const struct m0_conf_cache *cache,
+				 const struct m0_fid *id)
+{
+	struct m0_conf_obj *obj;
+	uint32_t cnt1;
+	uint32_t cnt2;
+        /*
+         * For process and service fid different approach is taken to compare
+	 * FID's. Specifically for process/service fid considering base fid only
+	 * (excluding counter bit) for comparison. Counter bits are higher 32
+	 * bist of the key.
+	 *  e.g. 
+	 *  |-------- Container------------| |----------------Key-------------|
+	 *
+	 *  F F F F F F F F F F F F F F F F : F F F F F F F F  F F F F F F F F
+	 *                                    |-Counter bits-|
+	 */
+	/*
+	 * TODO: Current implementation we are assuming that dynamic FID's will
+	 * always be generated in sequential manner (w.r.t counter bits), so
+	 * that we can fetch latest last dynamic FID, but It may not be in
+	 * sequential always. This scenario needs to be handle.
+	 */
+	if ((m0_fid_tget(id) == 'r') ||
+	    (m0_fid_tget(id) == 's')) {
+		m0_tl_for(m0_conf_cache, &cache->ca_registry, obj) {
+			if (m0_base_fid_eq(&obj->co_id, id)) {
+				cnt1 = obj->co_id.f_key >> 32 & 0xFFFFFFFF;
+				cnt2 = id->f_key >> 32 & 0xFFFFFFFF;
+				if (cnt1 == cnt2)
+					return obj;
+				else if (cnt1 != (cnt2 - 1))
```

Comment:
Suppose counter is at K(4) and after a restart of a m0d/motr client, new dynamic fid with counter 5 comes, it may not able to find the object with counter K, then it should proceed with last valid object of this process(upto base fid in reverse order), so return this object cached from here.